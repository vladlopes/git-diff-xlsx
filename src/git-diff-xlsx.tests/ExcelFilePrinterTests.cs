using System;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace git_diff_xlsx.tests
{
    [TestFixture]
    public class ExcelFilePrinterTests
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void CanPrintPlainTextCellValues()
        {
            var sut = new ExcelFilePrinter();
            using(var input = new MemoryStream(Resources.SingleSheetWithPlainTextValues))
            using(var output = new StringWriter())
            {
                sut.Print(input, output);
                Approvals.Verify(output.ToString());
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void CanPrintNumericCellValues()
        {
            var sut = new ExcelFilePrinter();
            using(var input = new MemoryStream(Resources.SingleSheetWithNumericValues))
            using(var output = new StringWriter())
            {
                sut.Print(input, output);
                Approvals.Verify(output.ToString());
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void CanPrintCalculatedCellValues()
        {
            var sut = new ExcelFilePrinter();
            using(var input = new MemoryStream(Resources.SingleSheetWithCalculatedValues))
            using(var output = new StringWriter())
            {
                sut.Print(input, output);
                Approvals.Verify(output.ToString());
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void CanPrintCalculatedAndFormattedCellValues()
        {
            var sut = new ExcelFilePrinter();
            using(var input = new MemoryStream(Resources.SingleSheetWithCalculatedAndFormattedValues))
            using(var output = new StringWriter())
            {
                sut.Print(input, output);
                Approvals.Verify(output.ToString());
            }
        }
    }
}
