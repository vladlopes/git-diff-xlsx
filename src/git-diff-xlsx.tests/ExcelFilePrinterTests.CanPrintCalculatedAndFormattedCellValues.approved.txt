Sheet1
File last edited by Mark Webb
=================================
Sheet: Sheet1 [ A1:C6 ]
=================================
    A1: Id
    B1: Name
    C1: Cost
    A2: 1
    B2: Item 1
    C2: £3.99
    A3: 2
    B3: Item 2
    C3: £2.95
    A4: 3
    B4: Item 3
    C4: £6.50
    B6: TOTAL
    C6: £13.44 SUM(C2:C5)
